libdnf/sack/query.cpp

```diff
@@ -1887,17 +1887,39 @@ Query::Impl::filterAdvisory(const Filter & f, Map *m, int keyname)
                 installed_solvables.push_back(pool_id2solvable(pool, installed_id));
             }
             std::sort(installed_solvables.begin(), installed_solvables.end(), NameArchSolvableComparator);
+
+            std::vector<Solvable *> obsolete_solvables;
+            Query obsoletes(sack, ExcludeFlags::IGNORE_EXCLUDES);
```

Comment:
Why did you use all available packages for `obsoletes` Query and not only available from `resultPset`?

Reply:
Oh right, we are only concerned with packages from `resultPset`. I have added a filter for that.